doc/design.md

```diff
@@ -243,3 +243,16 @@ For projects that specify a different source (typically code generated by the de
 An alternate solution special for deloboked code would be to replace the prefix of the generated directory with the original source directory. However then the line number would not fit, since Lombok expands the annotations. That could again be solved by searching in the original source for the error tag and by that determining the line number. However, we decided against this solution since for now the source position is not important enough.
 
 Needs: impl, utest
+
+
+### Maven Plugin should be Thread-Safe
+
+`dsn~mvn-plugin-thread-safe~1`
+
+In order to support parallel execution of the maven plugin and remove WARNING messages during parallel execution, maven plugin should be marked as thread-safe
+
```

Comment:
Please add a rationale why it's ok to mark it as thread-safe. Link to the Maven documentation and list for each criteria why it's fulfilled.
E.g.: "... Mojo does store state in static fields ..."